var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -0,0 +1,154 @@
+            args.append('-DCMAKE_CXX_COMPILER={0}'.format(self.spec['mpi'].mpicxx))
+            args.append('-DMPI_BASE_DIR:PATH={0}'.format(self.spec['mpi'].prefix))
+        else:
+            args.append('-DCMAKE_C_COMPILER={0}'.format(self.compiler.cc))
+            args.append('-DCMAKE_CXX_COMPILER={0}'.format(self.compiler.cxx))
+
+        # Currently FFTW_HOME and LIBXML2_HOME are used. CMake Warning is benign.
+        args.append('-DLIBXML2_HOME={0}'.format(self.spec['libxml2'].prefix))
+        args.append('-DLibxml2_INCLUDE_DIRS={0}'.format(self.spec['libxml2'].prefix.include))
+        args.append('-DLibxml2_LIBRARY_DIRS={0}'.format(self.spec['libxml2'].prefix.lib))
+        args.append('-DFFTW_HOME={0}'.format(self.spec['fftw'].prefix))
+        args.append('-DBOOST_ROOT={0}'.format(self.spec['boost'].prefix))
+        args.append('-DHDF5_ROOT={0}'.format(self.spec['hdf5'].prefix))
+        args.append('-DFFTW_INCLUDE_DIRS={0}'.format(self.spec['fftw'].prefix.include))
+        args.append('-DFFTW_LIBRARY_DIRS={0}'.format(self.spec['fftw'].prefix.lib))
+        
+        # Default is MPI, serial version is convenient for cases, e.g. laptops
+        if '~mpi' in self.spec:
+            args.append('-DQMC_MPI=0')
+
+        # when '-DQMC_CUDA=1', CMake automatically sets '-DQMC_MIXED_PRECISION=1'
+        # there is a double-precision CUDA path, but it is deprecated
+        if '+cuda' in self.spec:
+            args.append('-DQMC_CUDA=1')
+
+        # this is for the experimental mixed-prescision CPU code 
+        if '+mixed' in self.spec:
+            args.append('-DQMC_MIXED_PRECISION=1')
+
+        return args
+
+    def install(self, spec, prefix):
+        """Make the install targets"""
+        with working_dir(self.build_directory):
+            # QMCPACK 'make install' does nothing, which causes
+            # Spack to throw an error.  
+            #
+            # This install method creates the top level directory
+            # and copies the bin subdirectory into the appropriate
+            # location. We do not copy include or lib at this time due
+            # to technical difficulties.
+            mkdirp(prefix)
+
+            # install binaries
+            install_tree(join_path(self.build_directory, 'bin'),
+                         prefix.bin)
+
+            # install manual
+            install_tree(join_path(self.stage.source_path, 'manual'), 
+                         join_path(prefix,'manual'))
+
+            # install nexus
+            install_tree(join_path(self.stage.source_path, 'nexus'),
+                         join_path(prefix,'nexus'))
+
+    @run_after('build')
+    @on_package_attributes(run_tests=True)
+    def check_build(self):
+        """Run ctest after building binary. Use 'spack install --run-tests qmcpack'.
+        It can take 24 hours to run all the regression tests and unit tests. """
+        self.ctest
+        pass
```

Comment:
Ah, I see.

> As ctest should be able to take command line arguments which the Python module does not support.

Actually, it can take command-line arguments, and you don't need self:
```python
ctest('VERBOSE=ON')
```